dbcon/joblist/tuple-bps.cpp

```diff
@@ -2138,10 +2146,12 @@ void TupleBPS::sendPrimitiveMessages()
   }
 
 abort:
-  boost::unique_lock<boost::mutex> tplLock(tplMutex);
-  finishedSending = true;
-  condvar.notify_all();
-  tplLock.unlock();
+  {
+    boost::unique_lock<boost::mutex> lock(tplMutex);
+    finishedSending.store(true, std::memory_order_release);
+    condvar.notify_all();
+    condvarWakeupProducer.notify_all();
+  }
```

Comment:
nit: for my taste it would look better:
```cpp
  if (cancelled())
    return abort();
    
  try
  {
    makeJobs(&jobs);
    interleaveJobs(&jobs);
    sendJobs(jobs);
  }
  catch (...)
  {
    sendError(logging::ERR_TUPLE_BPS);
    handleException(std::current_exception(), logging::ERR_TUPLE_BPS, logging::ERR_ALWAYS_CRITICAL,
                    "st: " + std::to_string(fStepId) + " TupleBPS::sendPrimitiveMessages()");
    return abort();
  }

  boost::unique_lock<boost::mutex> lock(tplMutex);
  finishedSending.store(true, std::memory_order_release);
  condvar.notify_all();
  condvarWakeupProducer.notify_all();
```